int32 MOTION_TO_GOAL=           1
int32 MOTION_ODO_CALIB=         2
int32 MOTION_POTENTIAL_FIELD=   3
int32 MOTION_FOLLOW_PATH=       4
int32 MOTION_FOLLOW_TARGET=     5
int32 MOTION_FIXED_PARAMS=      6
int32 MOTION_FOLLOW_RSPATH=     7
int32 MOTION_DRIVE_PATTERN=     8
int32 MOTION_DIRECT_SPEED=      9
#goal definition
int32 mode
# pose of goal in WCS
float32 x
float32 y
float32 theta
nav_msgs/Path path
# topic to listen to receive path updates
string path_topic
# accepted tolerance of position
float32 pos_tolerance
# accepted tolerance of orientation in rad
# not implemented yet float32 theta_tolerance
# driving speed of robot (interpreted as recommended value)
float32 v
# sideslip angle of robot
float32 beta
# movement speed of goal (used in follow target)
float32 target_v
#int32 servof
#int32 servor
# fixed front steering angle
float32 deltaf
# fixed rear steering angle
float32 deltar
#float32 follow_dist
---
#result definition
int32 MOTION_STATUS_STOP=  0
int32 MOTION_STATUS_MOVING= 1
int32 MOTION_STATUS_COLLISION=2
int32 MOTION_STATUS_SUCCESS=3
int32 MOTION_STATUS_MOVE_FAIL=4
int32 MOTION_STATUS_INTERNAL_ERROR=5
int32 MOTION_STATUS_SLAM_FAIL=6
int32 MOTION_STATUS_GOAL_FAIL=7
int32 MOTION_STATUS_PATH_LOST=8
int32 mode
int32 status
float32 dist_driven
float32 dist_goal
---
#feedback
int32 MOTION_STOP   =        0
int32 MOTION_MOVING =        1
int32 MOTION_CALIB  =        2
int32 mode
int32 status
float32 dist_driven
float32 dist_goal
